lib/puppet/resource_api.rb

```diff
@@ -271,6 +273,34 @@ def refresh_current_state
         end
       end
 
+      def self.validate_title_attribute(resource_hash)
+        return unless resource_hash.key?(:title)
+        return if type_definition.namevars.size == 1
+
+        # Logic borrowed from Puppet::Resource.parse_title
```

Comment:
Initially I was not happy about this, but looking into the original code I'm a lot more OK with this much simpler version. Any changes to this code across puppet&rsapi already need to be in lockstep to avoid breaking anything.

Reply:
I wasn't super happy with it either, but it seemed the best option available and `parse_title` was marked `private`

Reply:
:+1: